casper/src/slowcooker/scala/coop.rchain.casper/HashSetCasperSpecification.scala

```diff
@@ -1,247 +1,229 @@
-//
-//    override def postCondition(state: State, result: Try[Result]): Prop = isValidAndNoErrors(result)
-//  }
-//
-//  case class DeployAndPropose(node: RNode) extends Command {
-//    override type Result = (ValidBlockProcessing, List[String])
-//
-//    override def run(sut: Sut): Result = {
-//      val validator = sut.nodes(node.idx).node
-//      val result = (deploy(validator, deployment(0)) >> create(validator) >>= (
-//          b => add(validator, b)
-//      )).result.right.get
-//      (result, validator.logEff.errors ++ validator.logEff.warns)
-//    }
-//
-//    override def nextState(state: State): State = state
-//
-//    override def preCondition(state: State): Boolean = !state(node.idx).deployed
-//
-//    override def postCondition(state: State, result: Try[Result]): Prop = isValidAndNoErrors(result)
-//  }
-//
-//  case class Receive(node: RNode) extends Command {
-//    override type Result = List[String]
-//
-//    override def run(sut: Sut): List[String] = {
-//      val validator = sut.nodes(node.idx).node
-//      validator.receive().result
-//      validator.logEff.errors ++ validator.logEff.warns
-//    }
-//
-//    override def nextState(state: State): State = state
-//
-//    override def preCondition(state: State): Boolean = true
-//
-//    override def postCondition(state: State, result: Try[Result]): Prop =
-//      result.isSuccess && result.get.isEmpty
-//  }
-//}
+package coop.rchain.casper
+
+import scala.util.{Random, Try}
+import cats.effect.Sync
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.dag.BlockDagStorage.DeployId
+import coop.rchain.blockstorage.syntax._
+import coop.rchain.casper.MultiParentCasper.ignoreDoppelgangerCheck
+import coop.rchain.casper.blocks.proposer.Created
+import coop.rchain.casper.genesis.Genesis
+import coop.rchain.casper.genesis.contracts._
+import coop.rchain.casper.helper.TestNode
+import coop.rchain.casper.helper.TestNode._
+import coop.rchain.casper.protocol.{BlockMessage, DeployData}
+import coop.rchain.casper.util.{ConstructDeploy, GenesisBuilder}
+import coop.rchain.catscontrib.TaskContrib._
+import coop.rchain.crypto.signatures.{Secp256k1, Signed}
+import coop.rchain.crypto.PublicKey
+import coop.rchain.models.BlockHash
+import coop.rchain.rholang.interpreter.util.RevAddress
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalacheck._
+import org.scalacheck.commands.Commands
+
+case class RNode(idx: Int, name: String, deployed: Boolean)
+
+object HashSetCasperProperties extends Properties("HashSetCasper") {
+  implicit def noShrink[T]: Shrink[T] = Shrink.shrinkAny
+  property("network") = HashSetCasperSpecification.property()
+}
+
+object HashSetCasperPropertiesApp {
+  def main(args: Array[String]): Unit =
+    HashSetCasperSpecification.property().check(_.withMinSuccessfulTests(1))
+}
+
+class NodeBox(val node: TestNode[Effect], var lastBlock: Option[BlockMessage]) {
```

Comment:
Just curious that can we use `val` instead of `var`?

Reply:
slowcooker is not executed I guess, and this is very old code. I think not worth looking into

Reply:
We can consider removing it later and come up with other plans for that purpose.

Reply:
> slowcooker is not executed I guess, and this is very old code. I think not worth looking into

We should be careful in assessing the existing code, just because it's old is not a reason to be ignored or abandon it.

Reply:
It was not compiling due to changes in block merge branch, so I commented that out temporarily. This PR adds minor fixes for it to compile. Might be I should use /* ... */ comments style for that, so its more clear what has been changed actually

Reply:
Yeah, block comments should be better in this case. I tried running this test on dev branch but it's failing there also. We need to check if it make sense to fix it or if it's not needed anymore.